dotnet/src/Connectors/Connectors.AssemblyAI/AudioToText/AssemblyAIAudioToTextService.cs

```diff
@@ -0,0 +1,292 @@
+    private readonly HttpClient _httpClient;
+
+    /// <summary>
+    /// Attributes is not used by AssemblyAIAudioToTextService.
+    /// </summary>
+    public IReadOnlyDictionary<string, object?> Attributes => new Dictionary<string, object?>();
+
+    /// <summary>
+    /// Creates an instance of the <see cref="AssemblyAIAudioToTextService"/> with an AssemblyAI API key.
+    /// </summary>
+    /// <param name="apiKey">OpenAI API Key</param>
+    /// <param name="httpClient"></param>
+    public AssemblyAIAudioToTextService(
+        string apiKey,
+        HttpClient httpClient
+    )
+    {
+        this._apiKey = apiKey;
+        this._httpClient = httpClient;
+    }
+
+    /// <summary>
+    /// Transcribe audio file.
+    /// </summary>
+    /// <param name="content">Audio content.</param>
+    /// <param name="executionSettings">The AI execution settings (optional).</param>
+    /// <param name="kernel">The <see cref="Kernel"/> containing services, plugins, and other state for use throughout the operation.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>Text content from audio content.</returns>
+    public async Task<TextContent> GetTextContentAsync(
+        AudioContent content,
+        PromptExecutionSettings? executionSettings = null,
+        Kernel? kernel = null,
+        CancellationToken cancellationToken = default)
+    {
+        string uploadUrl;
+        using var stream = content.Data!.ToStream();
+        {
+            uploadUrl = await this.UploadFileAsync(stream, cancellationToken).ConfigureAwait(false);
+        }
+
+        var transcriptId = await this.CreateTranscriptAsync(uploadUrl, executionSettings, cancellationToken)
+            .ConfigureAwait(false);
+        var transcript = await this.WaitForTranscriptToProcessAsync(transcriptId, cancellationToken)
+            .ConfigureAwait(false);
+
+        return new TextContent(
+            text: transcript.RootElement.GetProperty("text").GetString(),
+            modelId: null,
+            // TODO: change to typed object when AAI SDK is shipped
+            innerContent: transcript,
+            encoding: Encoding.UTF8,
+            metadata: null
+        );
+    }
+
+    /// <summary>
+    /// Transcribe audio file.
+    /// </summary>
+    /// <param name="fileStream">Stream of the audio file</param>
+    /// <param name="executionSettings">The AI execution settings (optional).</param>
+    /// <param name="kernel">The <see cref="Kernel"/> containing services, plugins, and other state for use throughout the operation.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns></returns>
+    public async Task<TextContent> GetTextContentAsync(
```

Comment:
If you want to introduce an additional method, you need to introduce it also in abstraction i.e. in the ITextToAudioService. Look how it is done in this closed PR https://github.com/microsoft/semantic-kernel/pull/5074.
But I think there is a better way, instead adding a stream to the AudioContent, create a new class StreamingAudioContent which will be descendant from the [StreamingKernelContent](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/SemanticKernel.Abstractions/Contents/StreamingKernelContent.cs). And next create overload in the ITextToAudioService with the StreamingAudioContent as parameter.

Reply:
I created `AudioStreamContent`. When looking at `StreamingKernelContent`, the class didn't feel relevant to this use case, but more relevant to streaming LLM results.